server/src/api/bill_exec.py

```diff
@@ -53,12 +53,35 @@ def create_bill(self):
         
         try:
             username = cherrypy.session["username"]
+            #Bill has a title
+            if ((len(title) < 1)):
+                raise ValueError("no title","title")
+            #Bill has a payer
+            if ((len(payer) < 1)):
+                raise ValueError("no payer","payer")
+            #Bill has a total greater than 0
+            if (float(total)<=0):
+                raise ValueError("bad total","total")
+            #Bill has someone who needs to pay their share
+            if (len(payment_objects) < 1):
+                raise ValueError("bill only has one person","split")
+            
             success = self.database.add_bill(payer,username,title,total,payment_objects)
         except KeyError as e:
             error_msg = "User not logged in"
```

Comment:
The `try` block has gotten a bit bigger, and it's no longer trivial to see that this `KeyError` always implies that the user has not logged in. Would it be clearer for the `username = cherrypy.session["username"]` line to have its own try block? Perhaps would it be better if this inner try block re-throws the `KeyError` to a custom error?